### PR TITLE
updated the models

### LINKED ISSUE #5: Add New Recommendation Models
**Description:**
Enhance the recommendation engine by integrating additional models to improve accuracy and personalization. Incorporate modern algorithms or frameworks to handle larger datasets and diverse recommendation criteria.

**Tasks:**

- Advanced recommendation systems.
- Implement and test new models in the backend (Python) for compatibility.
- Compare performance metrics (accuracy, response time) with existing models.
- Update the API to include options for selecting different recommendation models.

**Goal:**
Provide users with more accurate and diverse question recommendations by leveraging cutting-edge algorithms.

### PR DESCRIPTION
**Pull request by priyansh-narang2308**
Resolve #5 
Implemented both collaborative filtering and content-based filtering recommendation models
Even tried to integrate deep learning to it using the skleanr.
Utilized TensorFlow and PyTorch frameworks for efficient deep learning capabilities
Applied data preprocessing technique
Implemented metrics like F1 score, recall, and precision
Added new API parameter for model selection by making a parameter AS i said🫡🫡

### CODE REVIEW

**Comment by priyansh-narang2308** [comment]
@amri-tah Please can u see the changes

**Comment by priyansh-narang2308** [comment]
Hello @amri-tah Can u review the changes.🫡

**Comment by VishalTheHuman** [comment]
@priyansh-narang2308 explain why the model is better ???

**Comment by priyansh-narang2308** [comment]
Hey @VishalTheHuman 
The collabaritive filtering is used for the user similarities like the user can get the similar type of questions when hes solving 
the next is content-based filtering which like analyzes the questions based on difficulty , the topics 
Then even used deep learning like pytrch and tensorflow . this can help to capture the relationship bw user and its questions.
Then the data preprocessing which provids clean data
and at last the API parameter from which the user can select the best model suitable for them..
Thats why i used it ..

**Comment by VishalTheHuman** [comment]
@priyansh-narang2308 no need to change `backend/api` revert those changes. Also attach images on how your model works. Put your code only in `model` folder

**Comment by priyansh-narang2308** [comment]
Hey @VishalTheHuman As I am finding difficulty to solve this issue 
Can u pls unassign me this issue.. :(
